### PR TITLE
23.08 Release preparation

### PR DESCRIPTION
**Pull request by ktangsali**
This PR updates a few dependencies and fixes a few security issues.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Dependencies
 
+- Updated the base container to latest PyTorch base container which is based on torch 2.0
+- Container now supports CUDA 12, Python 3.10
+- symengine and vtk dependencies updated
```
Comment (akshaysubr) [inline CHANGELOG.md:29]: Can we list what version they've been updated to here?

**Review by akshaysubr — APPROVED** [review]

**Review by mnabian — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (ktangsali) [reply CHANGELOG.md]: Thanks! Added the details to the Changelog

**Pull request merged**